### DISCUSSION TITLE
Run a trigger sequence when already triggered

### DISCUSSION THREAD

**Discussion by luckyshot** [espanso/espanso#1216] [Feature requests and Ideas]
It would be really useful to be able to concatenate two triggers. Here's some examples:

```
,a       á
,aa      à
,aaa     ª
,e       é
,ee      è
,o       ó
,oo      ò
,ooo     ö
...

xD      😂 
xDD     🤣
```

aText, a similar software, can do this and it is very nice, it enables you to type accented languages such as Spanish, Italian, French, German, etc. with an English keyboard in a seamless way.

**Comment by federico-terzi** [comment]
Hey @luckyshot,

Thank you for filing the feature request! I'll make sure to investigate if this addition is possible :)

Cheers :)

**Comment by jaklan** [comment]
@federico-terzi I see 3 potential solutions here:
1) an option to add delay before expanding - so then you have some time to put additional chars. It could be either global option or per-pattern.
2) an option not to expand before hitting space - it should work nicely when typing e.g. `:phone1<space>`, `:phone2<space>`. It's quite similar how text replacement works natively in MacOS. Triggers incl. space like `:phone number 1` should also work then.
3) an option to confirm selection with enter - again MacOS-like behaviour, but it needs some pop-up with the replacement, which is currently not implemented (afaik).

**Comment by federico-terzi** [comment]
Hey @jaklan,

Thank you for the feedback!

> an option to add delay before expanding - so then you have some time to put additional chars. It could be either global option or per-pattern.

Yes, that might be an option

> an option not to expand before hitting space - it should work nicely when typing e.g. :phone1<space>, :phone2<space>. It's quite similar how text replacement works natively in MacOS. Triggers incl. space like :phone number 1 should also work then.

This one is coming in the next release (#594), as you will be able to specify `right_word: true`, which basically triggers a snippet only if it's followed by a word separator (similar to the [word](https://espanso.org/docs/matches/#word-triggers) option, but only considering the end)

> an option to confirm selection with enter - again MacOS-like behaviour, but it needs some pop-up with the replacement, which is currently not implemented (afaik).

This one would be trickier to implement over 3/4 platforms

So I'd say option 2 might be the best bet for this use-case, as it's already implemented in the new version. If we find it limited, we can investigate other ways!

Cheers :)

**Comment by jaklan** [comment]
@federico-terzi Great to hear! And I haven't noticed the `word` option before, that's the pretty cool one ❤️

**Comment by federico-terzi** [comment]
@jaklan no worries! Keep in mind that the `word` option is pretty buggy at the moment, but most problems will be solved in #594, so stay tuned!

**Comment by SpecialCharacter** [comment]
I have the same issue with zh and zh*. zh will trigger ž, but zh* will just result in ž*, not ź.
Is it possible to insert a "remind me" variable, so the unchanged "zh" is somewhere stored and can be modified into "zh*"?
(if remindMe = zh do xyz)

**Comment by SpecialCharacter** [comment]
For example, I want to add conditions:

- trigger: "h"
      if remindMe: z
        replace: "ž"
      elif remindMe: s
        replace: "š"

But it does not work, I do not know if I didn't get the "grammar" right or I have to wait for #594… any help is appreciated.

**Comment by federico-terzi** [comment]
@SpecialCharacter Thank you for the feedback! Currently, there is no way to specify conditions as you described :)

**Comment by SpecialCharacter** [comment]
Nooo... :( 
I need conditions.
I did some reading on Yaml, so I understand "if ... then" will not work.
What about boolean? Something like {remindMe: z} true --> replace: "ž"
Or is it possible to change the language of the default script?

**Comment by SpecialCharacter** [comment]
Any update? Or can you tell me an alternative program that has conditionality?

**Comment by federico-terzi** [comment]
No updates on this one, I had to focus my energies on the #594 release, after that we'll move on lower-priority issues :)

**Comment by jaklan** [comment]
@federico-terzi - just quick question. I have such a config:
```yaml
  - trigger: ":mail"
    replace: "my.email@gmail.com"
    word: true
```
Is there any way to automatically remove the whitespace I have to add at the end to trigger the shortcut? I know I can do sth like:
```yaml
  - trigger: ":mail"
    replace: "my.email@gmail.com$|$"
    word: true
```
but it doesn't remove the whitespace, so it breaks some input fields on the websites.

**Comment by federico-terzi** [comment]
> Is there any way to automatically remove the whitespace I have to add at the end to trigger the shortcut?

@jaklan You could use the following:

```yaml
  - trigger: ":mail "
    replace: "my.email@gmail.com"
```

The only limitation is that it will only trigger with a space at the end, and not with other "word delimiters", such as periods and commas. May I ask you why you choose the `word` option here? I would have used a regular "exact" match as follows, but I might be missing some context:

```yaml
  - trigger: ":mail"
    replace: "my.email@gmail.com"
```

Cheers :)

**Comment by jaklan** [comment]
@federico-terzi hah right, smart enough, thanks! And about the "why" - because I also have another triggers like ":mail2", ":mail3" etc. Ofc I could use ":2mail", ":3mail", it's just matter of personal taste and I prefer the former one.

**Comment by federico-terzi** [comment]
@jaklan I see! Thanks for sharing :)
